src/tools/perf/lib/ucp_tests.cc

```diff
@@ -418,17 +445,10 @@ class ucp_perf_test_runner {
             /* coverity[switch_selector_expr_is_constant] */
             switch (TYPE) {
             case UCX_PERF_TEST_TYPE_PINGPONG:
-                ptr = (volatile uint8_t*)buffer + length - 1;
-                while (*ptr != sn) {
-                    progress_responder();
-                }
+                wait_for_sn(buffer, sn);
```

Comment:
seems after change we read buffer head, before change we read buffer tail (+length-1)
pls also see comments on #7013

Reply:
yes. I'm not certain with this tail - AFAIU _length_ is length with contig datatype and iov count with iov.